src/utils/kamu-cli-puppet/src/kamu_cli_puppet_ext.rs

```diff
@@ -0,0 +1,228 @@
+use std::net::Ipv4Addr;
+use std::path::PathBuf;
+
+use async_trait::async_trait;
+use chrono::{DateTime, Utc};
+use datafusion::prelude::{ParquetReadOptions, SessionContext};
+use opendatafabric::serde::yaml::{DatasetKindDef, YamlDatasetSnapshotSerializer};
+use opendatafabric::serde::DatasetSnapshotSerializer;
+use opendatafabric::{DatasetID, DatasetKind, DatasetName, DatasetRef, DatasetSnapshot, Multihash};
+use serde::Deserialize;
+
+use crate::KamuCliPuppet;
+
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+#[async_trait]
+pub trait KamuCliPuppetExt {
+    async fn list_datasets(&self) -> Vec<DatasetRecord>;
+
+    async fn add_dataset(&self, dataset_snapshot: DatasetSnapshot);
+
+    async fn get_list_of_repo_aliases(
+        &self,
+        dataset_ref: &DatasetRef,
+    ) -> (Vec<String>, Vec<String>);
+
+    async fn complete<T>(&self, input: T, current: usize) -> Vec<String>
+    where
+        T: Into<String> + Send;
+
+    async fn start_api_server(self, e2e_data_file_path: PathBuf) -> ServerOutput;
+
+    async fn assert_last_data_slice(
+        &self,
+        dataset_name: &DatasetName,
+        expected_schema: &str,
+        expected_data: &str,
+    );
+}
+
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+#[async_trait]
+impl KamuCliPuppetExt for KamuCliPuppet {
+    async fn list_datasets(&self) -> Vec<DatasetRecord> {
+        let assert = self
+            .execute(["list", "--wide", "--output-format", "json"])
+            .await
+            .success();
+
+        let stdout = std::str::from_utf8(&assert.get_output().stdout).unwrap();
+
+        serde_json::from_str(stdout).unwrap()
+    }
+
+    async fn add_dataset(&self, dataset_snapshot: DatasetSnapshot) {
+        let content = YamlDatasetSnapshotSerializer
+            .write_manifest(&dataset_snapshot)
+            .unwrap();
+
+        let mut f = tempfile::NamedTempFile::new().unwrap();
+
+        f.as_file().write_all(&content).unwrap();
+        f.flush().unwrap();
+
+        self.execute(["add".as_ref(), f.path().as_os_str()])
+            .await
+            .success();
+    }
+
+    async fn get_list_of_repo_aliases(
+        &self,
+        dataset_ref: &DatasetRef,
+    ) -> (Vec<String>, Vec<String>) {
+        let assert = self
+            .execute([
+                "repo",
+                "alias",
+                "list",
+                dataset_ref.to_string().as_str(),
+                "--output-format",
+                "csv",
+            ])
+            .await
+            .success();
+
+        let stdout = std::str::from_utf8(&assert.get_output().stdout).unwrap();
+
+        stdout
+            .lines()
+            .skip(1) // Skip header
+            .fold((vec![], vec![]), |mut acc, line| {
+                // Skip name
+                let mut line_it = line.split(',').skip(1);
```

Comment:
Why not use serde_json here too?

Reply:
Unfortunately, `kamu repo alias list` does not support output to JSON.

Added a ticket:
- https://github.com/kamu-data/kamu-cli/issues/751